pkg/minikube/cluster/cluster.go

```diff
@@ -293,16 +296,54 @@ func engineOptions(config MachineConfig) *engine.Options {
 
 func createVirtualboxHost(config MachineConfig) drivers.Driver {
 	d := virtualbox.NewDriver(constants.MachineName, constants.Minipath)
-	d.Boot2DockerURL = config.MinikubeISO
+	d.Boot2DockerURL = config.GetISOCacheLocation()
 	d.Memory = config.Memory
 	d.CPU = config.CPUs
 	d.DiskSize = int(config.DiskSize)
 	return d
 }
 
+func (m *MachineConfig) CacheMinikubeISO() error {
+	// cache the miniube-iso to .minikube dir
+	response, err := http.Get(m.MinikubeISO)
+	if err != nil {
+		return err
+	} else {
+		out, err := os.Create(m.GetISOCacheLocation())
+		if err != nil {
+			return err
+		}
+		defer out.Close()
+		defer response.Body.Close()
+		_, err = io.Copy(out, response.Body)
+		if err != nil {
+			return err
+		}
+	}
+	m.MinikubeISO = "file://" + m.GetISOCacheLocation()
```

Comment:
cc @jimmidyson 
Now I wonder how this works on Windows. I think the filepath.Join will turn this into '\' characters, then we're appending on a file:// part to make it a URI. Maybe we do actually want to use '/' explicitly instead of os.Pathsep.